python/tank/util/shotgun.py

```diff
@@ -223,8 +223,16 @@ def __create_sg_connection(config_data=None, user=None):
         sg = shotgun_api3.Shotgun(
             config_data["host"],
             script_name=config_data["api_script"], api_key=config_data["api_key"],
-            http_proxy=config_data.get("http_proxy")
-        )
+            http_proxy=config_data.get("http_proxy"), connect=config_data.get("connect", True))
```

Comment:
The code below should already take care of this. But you're right,  I thought about automatically forcing `connect=False` if there's a value for `timeout_secs`. Thought that was a little too automatic and magic, but happy to change that.

It doesn't make much sense to set the timeout without `connect=False` since that value won't get picked up otherwise. So maybe that's better?